src/drivers/stm32/drv_io_timer.c

```diff
@@ -376,14 +380,39 @@ static int timer_set_rate(unsigned timer, unsigned rate)
 {
 
 	/* configure the timer to update at the desired rate */
-	rARR(timer) = 1000000 / rate;
+	rARR(timer) = timer_freq[timer] * 1000000 / rate;
```

Comment:
yeah, that was originally hardwired to 0xFFFF. Unwise attempt at being too general; needs some constraints added.